message/messaging_service.cc

```diff
@@ -573,6 +575,7 @@ static constexpr unsigned do_get_rpc_client_idx(messaging_verb verb) {
     case messaging_verb::HINT_MUTATION:
     case messaging_verb::HINT_SYNC_POINT_CREATE:
     case messaging_verb::HINT_SYNC_POINT_CHECK:
+    case messaging_verb::FORWARD_REQUEST:
         return 1;
```

Comment:
I think we need to allocate a new connection for this. Otherwise we run the risk of deadlocks.

Consider two nodes, A and B. Both are flooded with with COUNT(*) queries. They respond by sending FORWARD requests to each other. This happens, until the RPC servers on both nodes hit their capacity. The RPC servers process the FORWARD requests by issuing READ_DATA requests to each other. However, these READ_DATA requests are waiting for server resources. But the server resources will only be released when FORWARD processing is complete. So we deadlock. The deadlock might be resolved by timeouts, but that's not a good way out.

Allocating a new connection ensures the rpc server has a separate resource pool for FOWARD and READ_DATA requests, and so deadlock cannot occur. This is why MUTATION_DONE and MUTATION_FAILED requests have a separate connection too.